src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -557,10 +559,14 @@ public void onSubscribe(Flow.Subscription s) {
                         closed = this.closed;
                         if (!closed) {
                             this.subscription = s;
-                            assert buffers.remainingCapacity() > 1; // should contain at least 2
+                            // should contain at least 2
+                            assert buffers.remainingCapacity() > 1
+                                    : "buffers capacity: " + buffers.remainingCapacity()
+                                    + "closed: " + closed + "failed: " + failed;
```

Comment:
Hello Daniel, I think a space before `closed:` and `failed:` would be good. Other than that, this looks good.